common/components/AddressFieldFactory/AddressFieldDropdown.tsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { AppState } from 'reducers';
+import { translateRaw } from 'translations';
+import { setCurrentTo, TSetCurrentTo } from 'actions/transaction';
+import { AddressLabelPair } from 'actions/addressBook';
+import { getLabels } from 'selectors/addressBook';
+import { getToRaw } from 'selectors/transaction/fields';
+import { Address, Identicon } from 'components/ui';
+import './AddressFieldDropdown.scss';
+
+interface StateProps {
+  labels: ReturnType<typeof getLabels>;
+  currentTo: ReturnType<typeof getToRaw>;
+}
+
+interface DispatchProps {
+  setCurrentTo: TSetCurrentTo;
+}
+
+type Props = StateProps & DispatchProps;
+
+interface State {
+  activeIndex: number | null;
+}
+
+class AddressFieldDropdown extends React.Component<Props> {
+  public state: State = {
+    activeIndex: null
+  };
+
+  public componentDidMount() {
+    window.addEventListener('keydown', this.handleKeyDown);
+  }
+
+  public componentWillUnmount() {
+    window.removeEventListener('keydown', this.handleKeyDown);
+  }
+
+  public render() {
+    return this.getIsVisible() ? (
+      <ul className="AddressFieldDropdown" role="listbox">
+        {this.renderDropdownItems()}
+      </ul>
+    ) : null;
+  }
+
+  private renderDropdownItems = () =>
+    this.getFilteredLabels().map((filteredLabel: AddressLabelPair, index: number) => {
+      const { activeIndex } = this.state;
+      const { address, label } = filteredLabel;
+      const isActive = activeIndex === index;
+      const className = `AddressFieldDropdown-dropdown-item ${
+        isActive ? 'AddressFieldDropdown-dropdown-item--active' : ''
+      }`;
+
+      return (
+        <li
+          key={address}
+          className={className}
+          onClick={() => this.props.setCurrentTo(address)}
+          role="option"
+          title={`${translateRaw('SEND_TO')}${label}`}
+        >
+          <Identicon address={address} size="2rem" />
```

Comment:
On smaller screens the EthBlockies icon still gets squished 
<img width="574" alt="screen shot 2018-05-01 at 3 31 38 pm" src="https://user-images.githubusercontent.com/13836920/39489835-d0eec170-4d54-11e8-8f82-04ea941591ed.png">

Reply:
Whoops — I need to wrap those in containers with a minimum width. :+1:

Reply:
Fixed 👍 

<img width="331" alt="screen shot 2018-05-02 at 3 13 41 pm" src="https://user-images.githubusercontent.com/10479826/39546867-6de2bfec-4e1b-11e8-9b33-75181fd1b731.png">